File: front_end/lci_org_user_sdk.js
function get(keyword) {
    var reg = new RegExp("(^|&)" + keyword + "=([^&]*)(&|$)", "i");
    var r = window.location.search.substr(1).match(reg);
    if (r != null) return unescape(r[2]);
    return null;   //注意此处参数是中文，解码使用的方法是unescape ，那么在传值的时候如果是中文，需要使用escape('曲浩')方法来编码。
}

class Lci_user {
    RURL = '';
    OtherData = '';
    OtherID = '';
    OtherKey = '';
    SessionKey = null;

    reqapi(name, parameter, Call = function (data) {
        console.log(data)
    }) {
        // 请求API
        // name :API名字
        // parameter :参数
        // Call :请求后执行的函数,传入返回内容
        console.log(this.RURL + name);
        let url = this.RURL + name;
        parameter['SessionKey'] = this.SessionKey;
        $(function () {
            $.ajax({
                type: "GET",
                url: url,
                data: parameter,
                success: function (data) {
                    Call(data)
                    console.log(data);
                }
            });
        });
    }

    constructor(Lci_user_url = '/api/', SessionKey) {
        this.RURL = Lci_user_url;
        if (SessionKey == null) {
            this.SessionKey = $.cookie('SessionKey');
            ;
            if (this.SessionKey == null) {
                $(function () {
                    $.ajax({
                        type: "GET",
                        url: Lci_user_url + 'Get_session_key', //拼接接口地址
                        data: {},
                        success: function (data) {
                            $.cookie('SessionKey', data);
                            console.log(data);
                        }
                    });
                });
                // this.SessionKey = this.reqapi('Get_session_key',{})
            }
        }
    }

    signup(name, Key, nickname, postbox = None, Call = function (data) {
        console.log(data)
    }) {
        `
        注册:
        name 用户名
        key 密码(几乎没用)
        nickname 昵称
        postbox 邮箱(必须要)
        `
        if (postbox == None) {
            return 'The mailbox cannot be empty'
        }
        return this.reqapi('registered', {
            name: name,
            Key: Key,
            postbox: postbox,
            nickname: nickname,
        }, Call)
    }

    logon(id = null, name = null, postbox = null, Call = function (data) {
        console.log(data)
    }) {
        // 登录
        return this.reqapi('login_SendEmainVerification_API', {
                user_id: id,
                user_name: name,
                postbox: postbox,
            }, Call
        )
    }

    logonValidate(ValidateCode = null, Call = function (data) {
        console.log(data)
    }) {
        // 登录，验证验证码是否正确
        return this.reqapi('login_Email_Verifier_API', {
                Verification: ValidateCode,
            }, Call
        )
    }

    LoginStatus(Call = function (data) {
        console.log(data)
    }) {
        // 获取登录状态
        return this.reqapi('login_status', {}, Call
        )
    }
}

File: lci_org_user_sdk.py
import requests
import json


class Lci_user():
    SessionKey = None
    RURL = ''  # 服务器地址
    OtherData = ''  # 其他用户数据
    OtherID = 'LstCodeGuanli'  # 访问其他用户数据的ID
    OtherKey = 'y87238d8sny4832n8a7'  # 访问密码

    def reqapi(self, name, parameter: dict = {}):
        '''
        name :API name
        parameter :param
        '''
        kw = {'SessionKey': self.SessionKey}
        for k, v in parameter.items():
            kw[k] = None
            kw[k] = v
        return requests.post(f'{self.RURL}{name}', params=kw).text

    def __init__(self, Lci_user_url='http://lci-user.df100.ltd/api/', SessionKey=None):
        '''
        Lci_user_url :服务器地址
        SessionKey :SessionKey,可选,如果不传,就请求一个Key
        '''
        self.RURL = Lci_user_url
        self.SessionKey = SessionKey
        try:
            self.SessionKey = requests.get(self.RURL + 'Get_session_key',
                                           params={'SessionKey': self.SessionKey}).text  # 请求SessionKey
            print('Connection succeeded!')
            print('GET session key:{key}'.format(key=self.SessionKey))
        except:
            print('Failed to connect to server')

    def signup(self, name, Key, nickname, postbox=None):
        '''
        注册:
        name 用户名
        key 密码(几乎没用)
        nickname 昵称
        postbox 邮箱(必须要)
        '''
        if (postbox == None):
            raise Exception('The mailbox cannot be empty')
        return self.reqapi(name='registered', parameter={
            'name': name,
            'Key': Key,
            'postbox': postbox,
            'nickname': nickname,
        }).replace('OK, ID - > ', '')

    def logon(self, id=None, name=None, postbox=None):
        return self.reqapi(name='login_SendEmainVerification_API', parameter={
            'user_id': id,
            'user_name': name,
            'postbox': postbox,
        }).replace('OK, ID - > ', '')

    def logonValidate(self, ValidateCode):
        '''
        登录验证:
        ValidateCode :验证码
        '''
        return self.reqapi(name='login_Email_Verifier_API', parameter={
            'Verification': ValidateCode
        })

    def GetUserData(self):
        '''
        获得用户数据
        '''
        return self.UserData

    def PullingUserData(self):
        '''拉取服务器的用户数据做缓存'''
        self.UserData = json.loads(self.reqapi(name='get_user_data', parameter={}))

    def UploadOtherData(self):
        '''
        上传用户数据
        '''
        return self.reqapi(name='GetorChange_other_user_data', parameter={
            'key': self.OtherKey,
            'id': self.OtherID,
            'v': json.dumps(self.OtherData),
        })

    def PullingOtherData(self):
        '''
        拉取服务器的用户数据
        '''
        self.OtherData = json.loads(self.reqapi(name='GetorChange_other_user_data', parameter={
            'key': self.OtherKey,
            'id': self.OtherID,
        }))
        return self.OtherData

    def Change_user_data(self, data):
        '''更改服务器上的用户数据(昵称之类的)'''
        self.reqapi(name='Change_user_data', parameter={'data': json.dumps(data)})
